src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2304 @@
+    moving the SEM focus (as it's considered an axis in Odemis)
+    """
+
+    def __init__(self, name, role, parent, channel:str, **kwargs):
+        """
+        axes (set of string): names of the axes
+        """
+
+        self.channel = channel
+        fwd_info = parent.working_distance_info(channel=channel)
+        axes_def = {
+            "z": model.Axis(unit=fwd_info["unit"], range=fwd_info["range"]),
+        }
+
+        model.Actuator.__init__(self, name, role, parent=parent, axes=axes_def, **kwargs)
+
+        # will take care of executing axis move asynchronously
+        self._executor = CancellableThreadPoolExecutor(max_workers=1)  # one task at a time
+
+        # RO, as to modify it the server must use .moveRel() or .moveAbs()
+        self.position = model.VigilantAttribute({}, unit="m", readonly=True)
+        self._updatePosition()
+
+        if self.channel == "electron" and not hasattr(self.parent, "_scanner"):
+            raise ValueError("Required scanner child was not provided."
+                             "An ebeam scanner is a required child component for the Focus class")
+        if self.channel == "ion" and not hasattr(self.parent, "_fib_scanner"):
+            raise ValueError("Required fib scanner child was not provided."
+                             "An fib scanner is a required child component for the Focus class")
+
+        # Refresh regularly the position
+        self._pos_poll = util.RepeatingTimer(5, self._refreshPosition, "Focus position polling")
+        self._pos_poll.start()
+
+    @isasync
+    def applyAutofocus(self, detector):
+        """
+        Wrapper for running the autofocus functionality asynchronously. It sets the state of autofocus,
+        the beam must be turned on and unblanked. Also a a reasonable manual focus is needed. When the image is too far
+        out of focus, an incorrect focus can be found using the autofocus functionality.
+        This call is non-blocking.
+
+        :param detector: (model.Detector)
+            Detector component that stores information about which channel to use for autofocusing.
+        :return: Future object
+        """
+        # Create ProgressiveFuture and update its state
+        est_start = time.time() + 0.1
+        f = ProgressiveFuture(start=est_start,
+                              end=est_start + 11)  # rough time estimation
+        f._autofocus_lock = threading.Lock()
+        f._must_stop = threading.Event()  # cancel of the current future requested
+        f.task_canceller = self._cancelAutoFocus
+        f._channel_name = detector._scanner.channel
+        return self._executor.submitf(f, self._applyAutofocus, f)
+
+    def _applyAutofocus(self, future):
+        """
+        Starts autofocussing and checks if the autofocussing process is finished for ProgressiveFuture.
+        :param future (Future): the future to start running.
+        """
+        pass
+        # channel_name = future._channel_name
```

Comment:
If that is not supported, just remove entirely AutoFocus. Otherwise, make it work.